### ISSUE TITLE
Test Failure: HotThreadMonitorTest will occasionally fail

### ISSUE THREAD

**Issue by jakelandis** [elastic/logstash#8345]
I have seen this fail on more then 1 occasion:

```
18:19:40 org.logstash.instruments.monitors.HotThreadMonitorTest > testStackTraceSizeOption FAILED
18:19:40     java.lang.AssertionError: 
18:19:40     Expected: is <4>
18:19:40          but: was <1>
18:19:40         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
18:19:40         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
18:19:40         at org.logstash.instruments.monitors.HotThreadMonitorTest.lambda$testStackTraceSizeOption$6(HotThreadMonitorTest.java:57)
18:19:40         at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
18:19:40         at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
18:19:40         at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
18:19:40         at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
18:19:40         at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
18:19:40         at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
18:19:40         at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
18:19:40         at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
18:19:40         at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
18:19:40         at org.logstash.instruments.monitors.HotThreadMonitorTest.testStackTraceSizeOption(HotThreadMonitorTest.java:55)
```

**Comment by original-brownbear** [comment]
@jakelandis reproduced :)

<img width="850" alt="screen shot 2017-09-21 at 09 38 32" src="https://user-images.githubusercontent.com/6490959/30684229-aaf6803c-9eb0-11e7-8d9a-f5bb9897bfba.png">

**Comment by original-brownbear** [comment]
fixed in #8350
